proto/osmosis/gamm/v1beta1/osmosis.proto

```diff
@@ -24,17 +26,63 @@ message MsgSwapExactAmountIn {
   ];
 }
 
-message MsgSwapExactAmountInResponse {
-  string token_out_amount = 1 [
-
-    (gogoproto.customtype) = "cosmossdk.io/math.Int",
-    (gogoproto.moretags) = "yaml:\"token_out_amount\"",
-    (gogoproto.nullable) = false
-  ];
-}
-
 message SwapAmountInRoute {
   uint64 pool_id = 1 [ (gogoproto.moretags) = "yaml:\"pool_id\"" ];
   string token_out_denom = 2
       [ (gogoproto.moretags) = "yaml:\"token_out_denom\"" ];
 }
+
+// A TwapRecord stores the most recent price of a pair of denom's
+message OsmosisTwapRecord {
+  uint64 pool_id = 1;
+  // Lexicographically smaller denom of the pair
```

Comment:
Interesting, so the definition of asset0 vs asset1 is based on the alphabetic ordering? Presumably on the ibc denoms? We will have to be careful about this

Reply:
yup - we'll just have to sort them before submitting the request